### ISSUE TITLE
Performance loss when using iGibsonEnv

### ISSUE THREAD

**Issue by sycz00** [StanfordVL/iGibson#210]
Hey,

Iam looking for ways to improve the performance of my code and one thing I was wondering about, would it be faster using Simulator+ Scene import (+ render "seg" and "depth") manually as opposed to using the iGibsonEnv class ?

Thanks :)

**Comment by roberto-martinmartin** [comment]
yes, this may accelerate simulation! We do some extra processes within iGibsonEnv. The most important ones, the update of extended states (e.g., temperature, dirt level, wetness level...)., and the checking of logic predicates (is something on top of something? is something cooked?...). These are necessary for example for our BEHAVIOR tasks, which are defined in logic space.
If you do not need those, you may accelerate things without losing any relevant features. 
We may consider a "kinematics-only" version of the environment in the future for cases like yours.
